File: app/src/main/java/com/datarockets/mnchkn/store/GameService.java
package com.datarockets.mnchkn.store;

import com.datarockets.mnchkn.models.GameStep;
import com.datarockets.mnchkn.models.Player;

import java.util.List;
import java.util.Map;

import lecho.lib.hellocharts.model.LineChartData;

public interface GameService {
    void insertStep(Player player);
    void clearSteps();
    boolean isGameStarted();
    void createPlayerIdGameStepsMap();
    void setGameStatus(boolean gameStatus);
    Map<Player, List<GameStep>> getScoresChartData();
    LineChartData createScoresChartData(int type, Map<Player, List<GameStep>> playerGameStepsMap);
}

File: app/src/main/kotlin/com/datarockets/mnchkn/store/GameService.kt
package com.datarockets.mnchkn.store

import com.datarockets.mnchkn.models.GameStep
import com.datarockets.mnchkn.models.Player

import lecho.lib.hellocharts.model.LineChartData

interface GameService {
    fun insertStep(player: Player)
    fun clearSteps()
    val isGameStarted: Boolean
    fun createPlayerIdGameStepsMap()
    fun setGameStatus(gameStatus: Boolean)
    val scoresChartData: Map<Player, List<GameStep>>
    fun createScoresChartData(type: Int, playerGameStepsMap: Map<Player, List<GameStep>>): LineChartData
}
